### PR TITLE
feat(scroll): Add Scroll adapter

### PR DESCRIPTION
**Pull request by bmzig**
This instantiates a Scroll Adapter based off of the new generic adapter code. This is a smaller fix since I did not want to modify the Inventory/CrossChainTransfer clients when we would have to revert them back as soon as the generic code replaces the current.

### CODE REVIEW

File: src/clients/bridges/ScrollAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  ) {
+    const { SCROLL, MAINNET } = CHAIN_IDs;
+    const bridges = {};
+    const l2Signer = spokePoolClients[SCROLL].spokePool.signer;
+    const l1Signer = spokePoolClients[MAINNET].spokePool.signer;
+    SUPPORTED_TOKENS[SCROLL]?.map((symbol) => {
+      const l1Token = TOKEN_SYMBOLS_MAP[symbol].addresses[MAINNET];
+      const bridgeConstructor = CUSTOM_BRIDGE[SCROLL][l1Token] ?? CANONICAL_BRIDGE[SCROLL];
+      bridges[l1Token] = new bridgeConstructor(SCROLL, MAINNET, l1Signer, l2Signer, l1Token);
+    });
```
Comment (james-a-morris) [inline src/clients/bridges/ScrollAdapter.ts:20]: Do we do this elsewhere? Could be a candidate for pushing to a helper function.

Comment (bmzig) [reply src/clients/bridges/ScrollAdapter.ts:20]: We do it here: https://github.com/across-protocol/relayer/blob/0b64c1897bc735ea507ea5ea0235b96ceb43cbda/src/adapter/AdapterManager.ts#L42 in the new adapter manager. What do you think? Should this be factored out? I do like keeping everything together in the context of the Adapter manager constructor so we keep things easy to follow.

File: src/clients/bridges/ScrollAdapter.ts
Comment (pxrl) [inline src/clients/bridges/ScrollAdapter.ts:1]: This is pretty clever - nice way to shim the new adapter into the existing adapter manager!

**Review by pxrl — APPROVED** [review]

File: src/clients/bridges/ScrollAdapter.ts (hunk 1 above)
Comment (bmzig) [reply src/clients/bridges/ScrollAdapter.ts:20]: I'm going to keep it as is now and then if we do not fully migrate over to the generic adapters after a few more chains, then we will factor this out. This code _should_ not exist anyways come 1-2 months from now.

**Pull request merged**